### PR TITLE
Added Career model and join Collection CareerSubjects

### PR DESCRIPTION
**Pull request by frankmangone**
Probably lots of fixes to be done, but at the bare minimum, `Careers` can be created and `Subjects` can be added to them via a join collection `CareerSubjects`

### CODE REVIEW

File: src/entities/career/career.service.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    } else {
+      /* User must have admin permissons in the approving School
+       * For that, the School is added to the record that will be checked by CASL
+       */
+      const schoolDocument = await this.schoolService.getById(schoolId)
+      const school: School = plainToClass(
+        School,
+        (schoolDocument as any).toObject(),
+      )
+      career['approvingSchool'] = school
```
Comment (gastonmorixe) [inline src/entities/career/career.service.ts:71]: @frankmangone better to use a dot instead

**Review by gastonmorixe — APPROVED** [review]
LGTM :) minor comments then ok

File: src/entities/career/career.model.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import * as DB from '@nestjs/mongoose' // { Prop, Schema, SchemaFactory }
+import * as GQL from '@nestjs/graphql' // { Field, ObjectType, ID }
+import mongoose from 'mongoose'
+import autopopulate from 'mongoose-autopopulate'
+import * as Utils from '../../utils/Model'
```
[the remark this answers is not in the document]
Comment (frankmangone) [reply src/entities/career/career.model.ts:5]: Done!

File: src/entities/career/career.service.ts (hunk 1 above)
Comment (frankmangone) [reply src/entities/career/career.service.ts:71]: Sure, no problem

**Comment by frankmangone** [comment]
Closes issue #19

**Comment by frankmangone** [comment]
Closes issue #19

**Pull request merged**